web/src/routes/SearchPage.tsx

```diff
@@ -83,9 +84,9 @@ const SearchPage = ({ city, cityCode, languageCode, pathname }: CityRouteProps):
     )
   }
 
-  const handleFilterTextChanged = (filterText: string): void => {
-    setFilterText(filterText)
-    const appendToUrl = filterText.length !== 0 ? `?${SEARCH_QUERY_KEY}=${filterText}` : ''
+  const handleFilterTextChanged = (query: string): void => {
+    setSearchParams({ query })
+    const appendToUrl = query.length !== 0 ? `?query=${query}` : ''
     navigate(`${pathname}/${appendToUrl}`, { replace: true })
```

Comment:
🤔 I actually think these two lines are not necessary anymore as `setSearchParams` already updates the url, so you could just remove the whole function and use `query => setSearchParams({ query })` dierctly in L137 (or just keep the function but remove L89 and L90).